### PR TITLE
Add httpClientProvider and javascriptProvider to HttpHooksApi

### PR DESCRIPTION
**Pull request by fredrikhr**
# TL;DR

Adds the `httpClientProvider` and `javascriptProvider` singletons from `io` to the `HttpyacHooksApi` interface.

## `httpClientProvider`

If a plugin needs to do HTTP requests as part of its implementation it would be advantageous if the plugin could perform HTTP requests in the exact same way that Httpyac does. In doing so, the plugin would now no longer need to depend on an http-client, e.g. got, itself.

## `javascriptProvider`

When executing a script block in an HttpRegion the executed code gets access to a set of default modules (e.g. `httpyac` itself, `dayjs`, `vscode`, etc.). However, plugin authors do not get access to these modules. Moreover, some of these modules are in itself extendable (e.g. dayjs), but a plgin would have no direct means to access these.

Currently a plugin would need to call `utils.replaceVariables` and supply a JavaScript expression to obtain these modules.

### CODE REVIEW

**Comment by AnWeber** [comment]
thx
